src/obj_char_transfer.c

```diff
@@ -0,0 +1,872 @@
+    }
+    ObjCharTransferTask_Init(task);
+}
+
+static ObjCharTransferTask *ObjCharTransfer_GetTaskByID(int resId) {
+    for (int i = 0; i < sObjCharTransferTasksManager->max; ++i) {
+        if (sObjCharTransferTasksManager->tasks[i].resId == resId) {
+            return &sObjCharTransferTasksManager->tasks[i];
+        }
+    }
+
+    return NULL;
+}
+
+static GXOBJVRamModeChar ObjCharTransferTask_SetMappingTypeFromHW(ObjCharTransferTask *task, NNS_G2D_VRAM_TYPE vram) {
+    if (vram == NNS_G2D_VRAM_TYPE_2DMAIN) {
+        task->charData->mapingType = GX_GetOBJVRamModeChar();
+    } else {
+        task->charData->mapingType = GXS_GetOBJVRamModeChar();
+    }
+    return task->charData->mapingType;
+}
+
+static void ObjCharTransferTask_SetBaseAddrs(ObjCharTransferTask *task, u32 offsetMain, u32 offsetSub) {
+    task->baseAddrMain = offsetMain;
+    task->baseAddrSub = offsetSub;
+    if (!task->isTransferred) {
+        ObjCharTransferTask_G2dLoadImageMapping(task);
+    }
+}
+
+static BOOL getBlockNumAndFreeSpaceForTransfer(int vram, u32 *blockNumMain, u32 *blockNumSub, u32 size, u32 *freeSpaceMain, u32 *freeSpaceSub) {
+    if (vram & NNS_G2D_VRAM_TYPE_2DMAIN) {
+        *freeSpaceMain = ObjCharTransfer_AlignToBlock(size, sObjCharTransferTasksManager->blockSizeMain, TRUE);
+        u32 numBlocksUsed = ObjCharTransfer_CalcBlockNumLimit(*freeSpaceMain, sObjCharTransferTasksManager->blockSizeMain);
+        *blockNumMain = ObjCharTransfer_FindAvailableTransferBlock(numBlocksUsed, sObjCharTransferTasksManager->blockBufMain);
+        if (*blockNumMain == -1) {
+            GF_ASSERT(FALSE);
+            return FALSE;
+        }
+        *blockNumMain = ObjCharTransfer_GetBlockMemOffset(*blockNumMain, sObjCharTransferTasksManager->blockSizeMain);
+    }
+    if (vram & NNS_G2D_VRAM_TYPE_2DSUB) {
+        *freeSpaceSub = ObjCharTransfer_AlignToBlock(size, sObjCharTransferTasksManager->blockSizeSub, TRUE);
+        u32 numBlocksUsed = ObjCharTransfer_CalcBlockNumLimit(*freeSpaceSub, sObjCharTransferTasksManager->blockSizeSub);
+        *blockNumSub = ObjCharTransfer_FindAvailableTransferBlock(numBlocksUsed, sObjCharTransferTasksManager->blockBufSub);
+        if (*blockNumSub == -1) {
+            GF_ASSERT(FALSE);
+            return FALSE;
+        }
+        *blockNumSub = ObjCharTransfer_GetBlockMemOffset(*blockNumSub, sObjCharTransferTasksManager->blockSizeSub);
+    }
+    return TRUE;
+}
+
+static void ObjCharTransferTask_UpdateBaseAddrs(ObjCharTransferTask *task, u32 offsetMain, u32 offsetSub) {
+    if (task->vram & NNS_G2D_VRAM_TYPE_2DMAIN) {
+        task->baseAddrMain = offsetMain + sObjCharTransferTasksManager->freeSizeMain;
+    }
+    if (task->vram & NNS_G2D_VRAM_TYPE_2DSUB) {
+        task->baseAddrSub = offsetSub + sObjCharTransferTasksManager->freeSizeSub;
+    }
+}
+
+static void reserveTransferBlocksByVramOffsetAndSize(NNS_G2D_VRAM_TYPE vram, u32 offsetMain, u32 offsetSub, u32 sizeMain, u32 sizeSub) {
```

Comment:
prefix, declaration

Reply:
i say leave as-is, since it's a static function that does not act on any of this file's struct types

Reply:
ehh that is a fair point, however it is associated with ObjCharTransfer so imo it should have that prefix, even if it does not act on the struct itself